proposals/extensions.md

```diff
@@ -116,6 +109,251 @@ public implicit extension E for S
 }
 ```
 
+## Open issue: disambiguating between extension type method and classic extension method
+
+Should we disambiguate when signatures match?
+```csharp
+var x = new C().M; // ambiguous
+new C().M(); // ambiguous once we mix all extensions
+
+class C { }
+
+implicit extension E for C
+{
+    public static void M() { }
+}
+
+static class Extensions
+{
+    public static void M(this C c) { }
+}
+```
+
+## Open issue: removing zero-arity-matches-any from var scenario
+
+By the current rules, there are some unfortunate interactions between member access and natural function type.
```

Comment:
>there are some unfortunate interactions between member access and natural function type [](http://example.com/codeflow?start=22&length=87)

This is not specific. What interactions? Why might one consider them unfortunate?

Reply:
See the example below (line 142). A member that cannot work in the end gets in the way. We've taken steps in overload resolution and more recently in natural type to reduce such behaviors (discard bad candidates earlier).
I'm not sure we should do something here, maybe the current behavior is desirable fallout of current rules...

Reply:
I'm going to remove this open issue. I don't have a good proposal for how to make this scenario work and I'm no longer convinced it should.

Here's an updated summary of the question for the record:

## Open issue: removing zero-arity-matches-any from var scenario

By the current rules, a generic method which cannot possibly work in a function type scenario can hide
a member that would work:

```csharp
var x = C.Member; // error: member lookup finds C.Member (method group) and lacks type arguments to apply to that match

class C
{
    public static void Member<T>() { }
}

implicit extension E for C
{
    public static int Member = 42;
}
```

Is this a scenario we think should work?

For context, from the [member access rules](https://github.com/dotnet/csharpstandard/blob/draft-v8/standard/expressions.md#12871-general),
we perform a member lookup which includes generic methods even when no type arguments are present:
> [...] if a member lookup of I in E with K type parameters produces a match, then E.I is evaluated and classified as follows
> Note: When the result of such a member lookup is a method group and K is zero,
> the method group can contain methods having type parameters.
> This allows such methods to be considered for type argument inferencing. end note

See details from the [member lookup rules](https://github.com/dotnet/csharpstandard/blob/draft-v8/standard/expressions.md#1251-general):

> When K is zero, methods having type parameters are not removed, since the type inference process might be able to infer the type arguments.

See details on [natural type of method groups](https://github.com/dotnet/csharplang/blob/main/proposals/method-group-natural-type-improvements.md)